app/helpers/application_helper.rb

```diff
@@ -98,37 +98,35 @@ def byline(creation, options={})
   end
 
   def non_anonymous_byline(creation)
-    if creation.respond_to?(:author)
-      creation.author
-    else
-      pseuds = []
-      pseuds << creation.authors if creation.authors
-      pseuds << creation.pseuds if creation.pseuds && (!@preview_mode || creation.authors.blank?)
-      pseuds = pseuds.flatten.uniq.sort
-
-      archivists = {}
-      if creation.is_a?(Work)
-        external_creatorships = creation.external_creatorships.select {|ec| !ec.claimed?}
-        external_creatorships.each do |ec|
-          archivist_pseud = pseuds.select {|p| ec.archivist.pseuds.include?(p)}.first
-          archivists[archivist_pseud] = ec.author_name
+    Rails.cache.fetch("#{creation.cache_key}/byline-nonanon") do
+      if creation.respond_to?(:author)
+        creation.author
+      else
+        pseuds = []
+        pseuds << creation.authors if creation.authors
+        pseuds << creation.pseuds if creation.pseuds && (!@preview_mode || creation.authors.blank?)
+        pseuds = pseuds.flatten.uniq.sort
+
+        archivists = {}
+        if creation.is_a?(Work)
+          external_creatorships = creation.external_creatorships.select {|ec| !ec.claimed?}
+          external_creatorships.each do |ec|
+            archivist_pseud = pseuds.select {|p| ec.archivist.pseuds.include?(p)}.first
```

Comment:
Use `find` instead of `select.first`.<br>Space between { and | missing.<br>Space missing inside }.